src/pools/abstract-pool.ts

```diff
@@ -0,0 +1,173 @@
+import EventEmitter from 'events'
+import type { MessageValue } from '../utility-types'
+import type { IPool } from './pool'
+
+export type ErrorHandler<Worker> = (this: Worker, e: Error) => void
+export type OnlineHandler<Worker> = (this: Worker) => void
+export type ExitHandler<Worker> = (this: Worker, code: number) => void
+
+export interface IWorker {
+  on(event: 'error', handler: ErrorHandler<this>): void
+  on(event: 'online', handler: OnlineHandler<this>): void
+  on(event: 'exit', handler: ExitHandler<this>): void
+}
+
+export interface PoolOptions<Worker> {
+  /**
+   * A function that will listen for error event on each worker.
+   */
+  errorHandler?: ErrorHandler<Worker>
+  /**
+   * A function that will listen for online event on each worker.
+   */
+  onlineHandler?: OnlineHandler<Worker>
+  /**
+   * A function that will listen for exit event on each worker.
+   */
+  exitHandler?: ExitHandler<Worker>
+  /**
+   * This is just to avoid not useful warnings message, is used to set `maxListeners` on event emitters (workers are event emitters).
```

Comment:
```sugestion
   * This is just to avoid non-useful warning messages. Will be used to set `maxListeners` on event emitters (workers are event emitters).
```

But IMO this description is lacking some information on which warning messages are considered "non-useful` and what this task do to prevent them/what else happens in that case

Reply:
So you can attach only a max number of listeners that by default is 10.
https://nodejs.org/api/events.html#events_emitter_setmaxlisteners_n

We can maybe add this documentation to better explain that and also set it to Infinity why not?

Good point